playground/src/editor.tsx

```diff
@@ -116,38 +138,79 @@ export function Editor(props: {
     const code = editor.current?.getValue();
     if (code == null) return;
 
+    const config = {
+      sources: [["code", code]],
+      languageFeatures: ["preview-qir-gen"],
+      profile: profile,
+    } as ProgramConfig;
+
     if (props.activeTab === "ast-tab") {
-      props.setAst(await props.compiler.getAst(code, []));
+      props.setAst(
+        await props.compiler.getAst(
+          code,
+          config.languageFeatures ?? [],
+          config.profile,
+        ),
+      );
     }
     if (props.activeTab === "hir-tab") {
-      props.setHir(await props.compiler.getHir(code, []));
+      props.setHir(
+        await props.compiler.getHir(
+          code,
+          config.languageFeatures ?? [],
+          config.profile,
+        ),
+      );
+    }
+    const codeGenTimeout = 1000; // ms
+    if (props.activeTab === "qir-tab") {
+      let timedOut = false;
+      const compiler = props.compiler_worker_factory();
+      const compilerTimeout = setTimeout(() => {
+        log.info("Compiler timeout. Terminating worker.");
+        timedOut = true;
+        compiler.terminate();
+      }, codeGenTimeout);
+      try {
+        const qir = await compiler.getQir(config);
+        clearTimeout(compilerTimeout);
+        props.setQir(qir);
+      } catch (e: any) {
+        if (timedOut) {
+          props.setQir("timed out");
+        } else {
+          props.setQir(e.toString());
+        }
+      } finally {
+        compiler.terminate();
```

Comment:
Is it benign to call this twice? The setTimeout body on line 172 may have called it, then the finally here will call it also. If you look at `webviewPanel.ts` in the VS Code source, it uses a flag on timeout to ensure that terminate isn't called twice. (I'm guessing I did that for reason, any probably tested at the time, but I don't recall now).

Reply:
@minestarks Do you know if it is safe to call terminate multiple times?

Reply:
Should be easy enough to call it twice in the 'happy path' (i.e. duplicate line 185) and see if "bad stuff" happens.

Reply:
Seems to work fine. Called it 3x every time.